sematic/resolvers/resource_requirements.py

```diff
@@ -243,19 +245,69 @@ class KubernetesSecurityContext:
     capabilities: KubernetesCapabilities
 
 
+@dataclass(frozen=True)
+class KubernetesHostPathMount:
+    """A "hostPath"-type configuration for a volume to mount on the pod to allow access to
+    the underlying node's file system.
+
+    More details can be found here:
+    https://kubernetes.io/docs/concepts/storage/volumes/#hostpath
+
+    Attributes
+    ----------
+    node_path: str
+        The path on the underlying node to mount into the pod. Corresponds to the "path"
+        configuration.
+    pod_mount_path: str
+        The path where to mount the volume in the pod. Corresponds to the "mountPath"
+        configuration.
+    name: str
+        The name of the volume. Must be an RFC 1123-compliant max 64-character label.
+        Corresponds to the "name" configuration. If unspecified, or set as None or empty,
+        will default to a label that is auto-generated based on the `pod_mount_path`.
+    type: str
+        The type of the volume mount. Corresponds to the "type" configuration. Defaults to
+        the empty string.
+    """
+
+    node_path: str
+    pod_mount_path: str
+    name: str = field(default="")
+    type: str = field(default="")
+
+    def __post_init__(self):
+        if not self.name:
+            # from the kubernetes documentation:
+            # > a lowercase RFC 1123 label must consist of lower case alphanumeric
+            # > characters or '-', and must start and end with an alphanumeric character
+            # > (e.g. 'my-name', or '123-abc', regex used for validation is
+            # > '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
+            sanitized = re.sub("_|/|\\s", "-", self.pod_mount_path)
+            # pod_mount_path must be absolute, so we know sanitized starts with a dash
+            name = f"volume{sanitized}"
+            # must also have at most 64 characters
+            if len(name) > 64:
+                # randomize the last few characters to avoid artificial collisions caused
+                # by the truncation, when the user specifies multiple very long paths with
+                # the same root
+                name = f"{name[:58]}{uuid4().hex[:6]}"
+
+            object.__setattr__(self, "name", name)
```

Comment:
This violates the contract of a "frozen" dataclass. I would rather we have `__post_init__` raise an error for invalid names. We can have `name = field(default_factory=lambda: uuid.uuid4().hex)` or similar to provide a default name. We lose some degree of human-readableness but I think that's ok for this case.

Reply:
When you exit the `KubernetesHostPathMount(node_path=..., pod_mount_path=...)`, all the fields will be set and frozen. The only inconsistency is while the object is being instantiated, which should be viewed as an atomic operation anyway.

I would prefer to have the names of the mounts descriptive / informative instead of completely random.

Reply:
If it was intended to be supported, you'd probably be able to do `self.name = name` 😉 . But I won't quibble. ✅